x-pack/legacy/plugins/index_management/public/components/mappings_editor/reducer.ts

```diff
@@ -39,6 +41,10 @@ export interface State {
   documentFields: DocumentFieldsState;
   fields: NormalizedFields;
   fieldForm?: OnFormUpdateArg<any>;
+  fieldsJsonEditor: {
+    format(): MappingsFields;
+    isValid: boolean | undefined;
```

Comment:
The `isValid` is always boolean for the Json editor as we validate on each key stroke. As from my above comment, we could just type the `fieldsJsonEditor` like this:

```js
fieldsJsonEditor: Parameters<OnUpdateHandler>[0]
```